metsim/metsim.py

```diff
@@ -616,7 +682,12 @@ def wrap_run(func: callable, loc: dict, params: dict,
             t_end = None
 
         # Disaggregate to subdaily values
-        df_complete = disaggregate(df, params, sg, t_begin, t_end)
+        if params['prec_type'].upper() == 'TRIANGLE':
```

Comment:
Done.

Reply:
I have negated the need to do this check in all instances but one by simply adding the triangle method variables to the params dictionary instead of creating a separate dataframe of these variables. Now the only check is in the run function, and if the prec_type is triangle, it calls a new function "add_prec_tri_vars" to add these variables to the params dictionary. When I run pytest however, I get an error in the setup_output function with line 397 (see latest pushed version). Any suggestions on this? See the error message below:

 ``>               v = ', '.join(v) ``
 ``E               TypeError: sequence item 0: expected str instance, DataArray found ``

 ``metsim/metsim.py:398: TypeError ``